components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/api/SelfRegisterApi.java

```diff
@@ -98,7 +98,11 @@ public String mePostCall(SelfUserRegistrationRequest user) throws ApiException {
 
         List<Pair> localVarQueryParams = new ArrayList<Pair>();
 
-        Map<String, String> localVarHeaderParams = new HashMap<String, String>();
+        if (headers == null) {
+            headers = new HashMap<>();
+        }
+
+        Map<String, String> localVarHeaderParams = headers;
```

Comment:
Here we are modifying the received parameters (map of headers). This is wrong! .
can be changed to 
localVarHeaderParams = headers == null? new HashMap<>() : new HashMap(headers)